gcloud/resource_manager/project.py

```diff
@@ -0,0 +1,333 @@
+
+        data = {'name': self.name, 'labels': self.labels}
+        resp = client.connection.api_request(method='PUT', path=self.path,
+                                             data=data)
+        self.set_properties_from_api_repr(resp)
+
+    def exists(self, client=None):
+        """API call:  test the existence of a project via a ``GET`` request.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> project = client.project('purple-spaceship-456')
+            >>> project.exists()
+            False
+
+        You can also use the
+        :func:`gcloud.resource_manager.client.Client.get_project`
+        method to check whether a project exists, as it will return ``None``
+        if the project doesn't exist::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> print client.get_project('purple-spaceship-456')
+            None
+
+        See
+        https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/projects/get
+
+        :type client: :class:`gcloud.resource_manager.client.Client` or None
+        :param client: the client to use.  If not passed, falls back to
+                       the ``client`` attribute.
+        """
+        client = self._require_client(client=client)
+
+        try:
+            # Note that we have to request the entire resource as the API
+            # doesn't provide a way tocheck for existence only.
+            client.connection.api_request(method='GET', path=self.path)
+        except NotFound:
+            return False
+        else:
+            return True
+
+    def delete(self, client=None, reload_data=True):
+        """API call:  delete the project via a ``DELETE`` request.
+
+        See:
+        https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/delete
+
+        This actually changes the status (``lifecycleState``) from ``ACTIVE``
+        to ``DELETE_REQUESTED``.
+        Later (it's not specified when), the project will move into the
+        ``DELETE_IN_PROGRESS`` state, which means the deleting has actually
+        begun.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> project = client.get_project('purple-spaceship-123')
+            >>> project.delete()
+
+        :type client: :class:`gcloud.resource_manager.client.Client` or None
+        :param client: the client to use.  If not passed,
+                       falls back to the ``client`` attribute.
+
+        :type reload_data: bool
+        :param reload_data: Whether to reload the project with the latest
+                            state. If you want to get the updated status,
+                            you'll want this set to `True` as the DELETE
+                            method doesn't send back the updated project.
```

Comment:
Can you explicitly call out that it's the `lifecycleState` that matters most here (nothing else would change after a delete, right?).